internal/apps/builder/docker.go

```diff
@@ -14,139 +14,165 @@ import (
+	if !ok {
+		return fmt.Errorf("not a static site component")
+	}
 
-	server_name _;
-	server_tokens off;
+	lw := b.getLogWriter()
+	template.Render(lw, `{{success checkmark}} building static site image with built assets from {{highlight .}}{{nl 2}}`, c.GetOutputDir())
+	tmpDir, err := ioutil.TempDir("", "static-*")
+	if err != nil {
+		return fmt.Errorf("creating temporary build directory: %w", err)
+	}
+	defer func() {
+		_ = os.RemoveAll(tmpDir)
+	}()
 
-	root /www;
-	gzip_static on;
-}`)
+	err = os.WriteFile(filepath.Join(tmpDir, "nginx.conf"), []byte(b.getStaticNginxConfig()), 0664)
+	if err != nil {
+		return fmt.Errorf("writing nginx config: %w", err)
+	}
 
-		assetsCopyPath := b.component.(*godo.AppStaticSiteSpec).GetOutputDir()
-		dockerfileStatic.WriteString(fmt.Sprintf(`
-FROM %s	as content
-FROM nginx:alpine
+	dockerfile, buildArgs, err := b.staticSiteDockerfile()
+	if err != nil {
+		return err
+	}
+	err = os.WriteFile(filepath.Join(tmpDir, "Dockerfile.static"), dockerfile, 0644)
+	if err != nil {
+		return fmt.Errorf("writing static site config: %w", err)
+	}
 
-COPY nginx.conf /etc/nginx/conf.d/default.conf
-COPY --from=content %s /www
-		`, b.ImageOutputName(), assetsCopyPath))
+	// TODO: pull down staticSiteNginxImage if needed or too old
 
-		tar, err := archive.TarWithOptions(tmpDir, &archive.TarOptions{})
-		if err != nil {
-			return res, err
-		}
-		dockerRes, err := b.cli.ImageBuild(ctx, tar, types.ImageBuildOptions{
-			Dockerfile: "./Dockerfile.static",
-			Tags: []string{
-				b.ImageOutputName() + "-static",
-			},
-		})
-		if err != nil {
-			res.ExitCode = 1
-			return res, err
-		}
-		defer dockerRes.Body.Close()
-		print(dockerRes.Body, lw)
+	tar, err := archive.TarWithOptions(tmpDir, &archive.TarOptions{})
+	if err != nil {
+		return fmt.Errorf("preparing build context: %w", err)
+	}
+	res, err := b.cli.ImageBuild(ctx, tar, dockertypes.ImageBuildOptions{
+		Dockerfile: "./Dockerfile.static",
+		Tags:       []string{b.StaticSiteImageOutputName()},
+		BuildArgs:  buildArgs,
+	})
+	if err != nil {
+		return err
 	}
+	defer res.Body.Close()
+	err = print(res.Body, lw)
+	fmt.Fprint(lw, "\n")
+	if err != nil {
+		return err
+	}
+	return nil
+}
 
-	res.Image = b.ImageOutputName()
-	res.BuildDuration = time.Since(start)
-	res.ExitCode = 0
-	return res, nil
+func (b *DockerComponentBuilder) staticSiteDockerfile() (dockerfile []byte, buildArgs map[string]*string, err error) {
+	c, ok := b.component.(*godo.AppStaticSiteSpec)
+	if !ok {
+		return nil, nil, fmt.Errorf("not a static site component")
+	}
+
+	dockerfile = []byte(`
+ARG app_image
+ARG nginx_image
+ARG output_dir
+FROM ${app_image} as content
+FROM ${nginx_image}
+ARG output_dir
+
+COPY nginx.conf /etc/nginx/conf.d/default.conf
+COPY --from=content ${output_dir}/ /www
+`)
+
+	buildArgs = map[string]*string{
+		"app_image":   strPtr(b.AppImageOutputName()),
+		"nginx_image": strPtr(staticSiteNginxImage),
+		"output_dir":  strPtr(c.GetOutputDir()),
```

Comment:
Do we need to join this with the workspace path similar to the cnb asset path calculation?

Reply:
nope, `output_dir` must be an absolute path for dockerfile builds. we are missing that validation rule though—I'll add it

Reply:
https://github.com/digitalocean/doctl/pull/1230/commits/7fa0c2a8eff9d634af4a10e30048ae004a78fbcb